src/Service.Tests/Unittests/ConfigFileWatcherUnitTests.cs

```diff
@@ -159,5 +161,109 @@ public void HotReloadConfigRestRuntimeOptions()
+        /// Internally, ConfigFileWatcher.NewFileContentsDetected is raised when the hash
+        /// of the file contents differ from what was previously detected.
+        /// In this test, the file contents are the same: MyValue -> MyValue.
+        /// The config file name is specific to this test in order to avoid concurrently
+        /// running tests from stepping over eachother due to acquiring a handle(s) to the file.
+        /// </summary>
+        [TestMethod]
+        public void ConfigFileWatcher_NotifiedOfZeroNetNewChange()
+        {
+            // Arrange
+            int fileChangeNotificationsReceived = 0;
+            string configName = "ConfigFileWatcher_NotifiedOfZeroNetNewChange.json";
+
+            // Write initial value to file.
+            File.WriteAllText(configName, "MyValue");
+
+            ConfigFileWatcher fileWatcher = new(directoryName: Directory.GetCurrentDirectory(), configFileName: configName);
+            fileWatcher.NewFileContentsDetected += (sender, e) =>
+            {
+                // For testing, modification of fileChangeNotificationsRecieved
+                // should be atomic. 
+                Interlocked.Increment(ref fileChangeNotificationsReceived);
+            };
+
+            // Act - Write to the file multiple times with the same value to implicitly trigger
+            // multiple file change events for the ConfigFileWatcher's filesystem watcher.
+            ModifyConfigFile(configFileName: configName, configContent: "MyValue");
+
+            // Assert -> allow time for the file change events to be processed.
+            // Wait time is arbitrary.
+            Thread.Sleep(millisecondsTimeout: 500);
+            Assert.AreEqual(expected: 0, actual: fileChangeNotificationsReceived);
+        }
+
+        /// <summary>
+        /// Wrapper around File.WriteAllText() such that IOExceptions result in retries
+        /// to allow for other handles on file to be dropped.
+        /// RunCount and sleep time are arbitrary.
+        /// </summary>
+        /// <param name="configFileName">Name of write-target config file.</param>
+        /// <param name="configContent">Content to write to file.</param>
+        private static void ModifyConfigFile(string configFileName, string configContent)
+        {
+            int runCount = 1;
+            while (runCount < 4)
+            {
+                try
+                {
+                    File.WriteAllText(configFileName, configContent);
+                    return;
+                }
+                catch (IOException ex)
+                {
+                    Console.WriteLine($"IO Exception, retrying due to {ex.Message}");
+                    if (runCount == 3)
+                    {
+                        throw;
+                    }
+
+                    // Constant backoff because we don't want this to hold up the CI/CD pipelines.
+                    // Wait time is arbitrary.
```

Comment:
Arbitrary wait times make this test non-deterministic.. Is there any way to make this more deterministic? Perhaps, write again only after we are sure of not getting an open handle exception?

Reply:
Found a way to utilize `TestableIO.System.IO.Abstractions` like we do elsewhere to mock the filesystem and filewatcher objects. Then I was able to manually induce the file-changed event instead of waiting for the event to be raised by the filesystem directly (which as you correctly point out, leads to possible flakiness due to depending on filesystem 'eventually' raising the file changed event). 
In a real hot-reload scenario, I am keeping the 'exponential' backoff in the utility which calculates file content hash because there may actually be IOExceptions when files are inaccessible due to other processes. However for testing, we want quick and determinate.